apptest/tests/maxingestionrate_test.go

```diff
@@ -0,0 +1,35 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/apptest"
+)
+
+func TestSingleMaxIngestionRateIncrementsMetric(t *testing.T) {
+	tc := apptest.NewTestCase(t)
+	defer tc.Stop()
+	sut := tc.MustStartVmsingle("vmsingle", []string{"-maxIngestionRate=1"})
+	sut.PrometheusAPIV1ImportPrometheus(t, docData, apptest.QueryOpts{})
+	maxIngestionRateMetric := sut.GetMetric(t, "vm_max_ingestion_rate_limit_reached_total")
+	if maxIngestionRateMetric <= 0 {
+		t.Errorf("Max Ingestion Rate metric not set unexpectedly")
+	} else {
+		t.Logf("MAX INGEST RATE HIT %d times", int(maxIngestionRateMetric))
+	}
```

Comment:
Please try following got/want pattern, for example:

```
if got, want  := sut.GetIntMetric(t, "vm_max_ingestion_rate_limit_reached_total"), 1; got != want {
  t.Fatalf("Unexpected vm_max_ingestion_rate_limit_reached_total: got %d, want %d", got, want)
}
```

Reply:
not sure how I would make this pattern fit here when got needs be > 0 - want is variable here.  Suggestions?

Reply:
It will, provided that the total number cannot be known in advance. But if the total number expected to always be the same (such as 1) I suggest using it.